### PR TITLE
feat: support no gas usage with ganache

### PR DESCRIPTION
**Pull request by bilbeyt**
### What I did

Added more ganache cli options 

fixes: APE-514

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by bilbeyt** [comment]
this is now ready for review @fubuloubu @unparalleled-js

**Comment by unparalleled-js** [comment]
> this is now ready for review @fubuloubu @unparalleled-js 

I am in the process of review and testing and will update shortly!

**Review by unparalleled-js — COMMENTED** [review]
There is still something I need to check, but I figure I will leave these comments for now

File: ape_ganache/provider.py [hunk 1]
```diff
@@ -33,6 +34,18 @@
 GANACHE_CHAIN_ID = 1337
 
 
+class Hardforks(Enum):
```
[the remark this answers is not in the document]
Comment (bilbeyt) [reply ape_ganache/provider.py:37]: I got these from ganache-cli version > 7, i am not sure if they are also there for geth or some another eth providers.

[the remark this answers is not in the document]
Comment (bilbeyt) [reply ape_ganache/provider.py:37]: will create sections for config.

[the remark this answers is not in the document]
Comment (bilbeyt) [reply ape_ganache/provider.py:37]: @unparalleled-js created the sections.

**Review by unparalleled-js — DISMISSED** [review]

File: ape_ganache/provider.py [hunk 2]
```diff
@@ -42,6 +55,14 @@ class GanacheForkConfig(PluginConfig):
     block_number: Optional[int] = None
 
 
+class GanacheMinerConfig(PluginConfig):
+    gas_price: int = 2000000000
```
Comment (unparalleled-js) [inline ape_ganache/provider.py:59]:
```suggestion
    gas_price: int = 2_000_000_000
```

**Review by unparalleled-js — COMMENTED** [review]
I did this locally and it worked.
In the file `tests/ape-config.yaml`, add this section to the `ganache` part:

```
  miner:
    gas_price: 2111222333
```

and then in `tests/test_provider.py::test_gas_price`, change it to be like this:

```python
def test_gas_price(connected_provider):
    gas_price = connected_provider.gas_price
    assert gas_price == 2_111_222_333  # Value set in `ape-config.yaml` file.
```

**Review by unparalleled-js — APPROVED** [review]

**Pull request merged**